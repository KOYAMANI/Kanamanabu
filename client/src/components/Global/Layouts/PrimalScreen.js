import React from 'react'
import { Container, Row } from 'react-bootstrap'
import { 
    Divider,
    Heading,
    PrimalBackground,
    Page,
 } from './PrimalScreenElements'

const PrimalScreen = ({title, children}) => {
    return (
        <PrimalBackground>
        <Container>
          <Row>
            <Page>
              {title && (
                <>
                  <Heading>{title}</Heading>
                  <Divider />
                </>
              )}
              {children}
            </Page>
          </Row>
        </Container>
      </PrimalBackground>
       
    )
}

export default PrimalScreen
