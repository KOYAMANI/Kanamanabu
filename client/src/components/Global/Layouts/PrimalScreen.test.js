import {render, screen, cleanup} from '@testing-library/react';
import ReactDOM from "react-dom";
import renderer from 'react-test-renderer'
import PrimalScreen from "./PrimalScreen";

afterEach(cleanup)

describe('ErrorMessage', () =>{
    test('Render without crashing', ()=>{
        const div = document.createElement('div')
        ReactDOM.render(<PrimalScreen title="primal-screen-test" >primal-screen-children</PrimalScreen>, div)
    });

    test('Render text', ()=>{
        render(<PrimalScreen title="primal-screen-test" >primal-screen-children</PrimalScreen>)
        const element = screen.getByText('primal-screen-test')
        const children = screen.getByText('primal-screen-children')
        expect(element).toBeInTheDocument();
        expect(element).toHaveTextContent('primal-screen-test')
        expect(children).toHaveTextContent('primal-screen-children')
    })

    test('match snapshot',()=>{
        const tree = renderer.create(<PrimalScreen title="primal-screen-test" >primal-screen-children</PrimalScreen>).toJSON();
        expect(tree).toMatchSnapshot();
    })
})