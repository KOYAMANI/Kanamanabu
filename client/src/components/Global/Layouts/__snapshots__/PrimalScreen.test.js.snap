// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage match snapshot 1`] = `
<div
  className="sc-bczRLJ cQsTxG"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="sc-dkzDqf bxnkfQ"
      >
        <div
          className="sc-hKMtZM kmgeex"
        >
          primal-screen-test
        </div>
        <hr
          className="sc-gsnTZi jwWTyC"
        />
        primal-screen-children
      </div>
    </div>
  </div>
</div>
`;
